x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/InferenceConfig.java

```diff
@@ -65,6 +74,39 @@ default boolean supportsSearchRescorer() {
         return false;
     }
 
+    @Nullable
+    default TrainedModelInput getDefaultInput(TrainedModelType modelType) {
+        if (modelType == null) {
+            return null;
+        }
+        return modelType.getDefaultInput();
+    }
+
+    default ActionRequestValidationException validateTrainedModelInput(
```

Comment:
to me it does not make much sense to have this method in here - its implementation seems to highly depend of what happens in `TrainedModelConfig`. Sending in the `validationException` also seems like a code smell.
Furthermore I don't see why it's an instance method - it does not depend on any of the instance variables, it could even be static?

Instead of moving this code from `TrainedModelConfig` - I would have added a simple `boolean requiresInputFields` method here that returns true and override it in `LearningToRankConfig`.
Then we could just use this method in `TrainedModelConfig` to do the validation.
With this approach we also don't need to add the `getDefaultInput` method to `InferenceConfig`.

Reply:
I can remember another PR with the exact opposite point of view from Ben in a similar case (see comment https://github.com/elastic/elasticsearch/pull/109071#discussion_r1618802705).